tools/ALARAJOYWrapper/preprocess_fendl3.py

```diff
@@ -14,6 +13,49 @@ def args():
     )
     return parser.parse_args()
 
+def write_dsv(dsv_path, cumulative_data):
+    """
+    Write out a space-delimited DSV file from the list of dictionaries,
+        dsv_path, produced by iterating through each reaction of each isotope
+        to be processed. Each row in the resultant DSV file is ordered as such:
+
+            pKZA dKZA emitted_particles non_zero_groups xs_1 xs_2 ... xs_n
+
+        Each row can have different lengths, as only non-zero cross-sections
+        are written out. The file is sorted by ascending parent KZA value.
+
+    Arguments:
+        dsv_path (str): Filepath for the DSV file to be written.
+        cumulative_data (list of dicts): List containing separate dictionaries
+            for each reaction contained in all of the TENDL/PENDF files
+            processed.
+
+    Returns:
+        None 
+    """
+
+    xs_key = 'Cross Sections'
+    join_keys = list(cumulative_data[0].keys())
+    join_keys.remove(xs_key)
+    
+    # Sort list of reaction dictionaries by ascending parent KZAs
+    cumulative_data.sort(key=lambda rxn: rxn.get(join_keys[0], float('inf')))
```

Comment:
My assumption is that it was already sorted into blocks of the same parent KZA, but those parent KZAs are ordered based on the alphabetical ordering of the element symbols?